src/Maestro/maestro-angular/src/app/services/channel.service.ts

```diff
@@ -45,6 +48,19 @@ export class ChannelService {
     return this.buildRepositoriesList(channelId);
   }
 
+  public getFlowGraph(channelId: number): Observable<StatefulResult<FlowGraph>> {
+    return of(channelId).pipe(
```

Comment:
This is a pretty odd method.  Why the "of(channelId)".  can't this method just be
```
if (channelId in ChannelService.graphCache) {
          return of(ChannelService.graphCache[channelId ]);
        }
        return this.maestro.channels.getFlowGraphAsync({channelId }).pipe(
          tap(graph => ChannelService.graphCache[channelId ] = graph),
        );
```

Reply:
I followed the pattern that we did for getBuildGraph in build.service.ts. I'll try your way and get back to you.

Reply:
Ok, so apparently, if you do it this way you get the message: 

> Argument of type '{ channelId: number; }' is not assignable to parameter of type '{ id: number; }'.
  Object literal may only specify known properties, and 'channelId' does not exist in type '{ id: number; }'.

Reply:
renaming the parameter